### ISSUE TITLE
[Bug]: Issue run `bundle exec jekyll serve`

### ISSUE THREAD

**Issue by TheRealOne78** [jekyll/jekyll#9642]
### Operating System

Artix Linux x86_64

### Ruby Version

ruby 3.0.6p216 (2023-03-30 revision 23a532679b) [x86_64-linux]

### Jekyll Version

jekyll 4.3.3

### GitHub Pages Version

Latest

### Expected Behavior

I tried creating a simple Jekyll site by following the instructions on https://jekyllrb.com/docs/installation/other-linux/ and https://jekyllrb.com/docs/. I also took notice of the failure note, but it still does not work.

### Current Behavior

It seems like the bundle tool is trying to install the dependencies globally instead of the local path (see logs).

I tried fixing this issue with ChatGPT, but even it doesn't seem to aknowledge why this happens. In the end, we ended up setting the bundler to install everything localy through a ~/.bundle/config file and it did install some dependencies, but then jekyll also ended up giving an exception, stating that json is not installed. I'm not even at a begginer level, so maybe this is a me-issue, a bug with jekyll, ruby stuff, or the docs aren't clear enough.

### Relevant log output

```shell
❯ bundle add webrick
Fetching gem metadata from https://rubygems.org/............
Resolving dependencies...
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Fetching strscan 3.1.0

Retrying download gem from https://rubygems.org/ due to error (2/4): Bundler::PermissionError There was an error while trying to write to `/usr/lib/ruby/gems/3.0.0/cache/strscan-3.1.0.gem`. It is likely that you need to grant write permissions for that path.

Retrying download gem from https://rubygems.org/ due to error (3/4): Bundler::PermissionError There was an error while trying to write to `/usr/lib/ruby/gems/3.0.0/cache/strscan-3.1.0.gem`. It is likely that you need to grant write permissions for that path.

Retrying download gem from https://rubygems.org/ due to error (4/4): Bundler::PermissionError There was an error while trying to write to `/usr/lib/ruby/gems/3.0.0/cache/strscan-3.1.0.gem`. It is likely that you need to grant write permissions for that path.

Bundler::PermissionError: There was an error while trying to write to `/usr/lib/ruby/gems/3.0.0/cache/strscan-3.1.0.gem`. It is
likely that you need to grant write permissions for that path.
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/shared_helpers.rb:107:in `rescue in
filesystem_access'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/shared_helpers.rb:104:in `filesystem_access'
/home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/rubygems_integration.rb:445:in `block in
download_gem'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/retry.rb:40:in `run'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/retry.rb:30:in `attempt'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/rubygems_integration.rb:437:in `download_gem'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/source/rubygems.rb:489:in `download_gem'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/source/rubygems.rb:446:in `fetch_gem'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/source/rubygems.rb:430:in `fetch_gem_if_possible'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/source/rubygems.rb:162:in `install'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/installer/gem_installer.rb:54:in `install'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/installer/parallel_installer.rb:132:in `do_install'
/home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/installer/parallel_installer.rb:123:in `block in
worker_pool'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/worker.rb:62:in `apply_func'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/worker.rb:57:in `block in process_queue'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/worker.rb:54:in `loop'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/worker.rb:54:in `process_queue'
  /home/therealone/.local/share/gem/ruby/3.0.0/gems/bundler-2.5.16/lib/bundler/worker.rb:90:in `block (2 levels) in create_threads'

An error occurred while installing strscan (3.1.0), and Bundler cannot continue.

In Gemfile:
  minima was resolved to 2.5.1, which depends on
    jekyll-feed was resolved to 0.17.0, which depends on
      jekyll was resolved to 4.3.3, which depends on
        kramdown-parser-gfm was resolved to 1.1.0, which depends on
          kramdown was resolved to 2.4.0, which depends on
            rexml was resolved to 3.3.2, which depends on
              strscan
```

### Steps to reproduce this issue:
1. sudo pacman -Syu ruby base-devel
2. gem install jekyll bundler
3. jekyll new myblog
4. cd myblog
5. bundle exec jekyll serve # fail
6. bundle add webrick  # fail

At some point, I also added `$HOME/.local/share/gem/ruby/3.0.0/bin` in my PATH, in my .zshrc file, because gem installs jekyll there:
```
export PATH="$HOME/.local/share/gem/ruby/3.0.0/bin:$PATH"
```

**Comment by TheRealOne78** [comment]
I seem to have found some sort of workaround:

1. Write the following in .bashrc or .zshrc:
```sh
# Ruby gem
export GEM_HOME="$(ruby -e 'puts Gem.user_dir')"
export PATH="$PATH:$GEM_HOME/bin"
``` 
2. Close and start a new shell session
3. gem install jekyll bundle
4. jekyll new sitename
5. cd sitename
6. bundle add json
7. bundle exec jekyll serve

The server works, but I get the following errors:
```trace
Configuration file: /home/therealone/git/therealone78.ddns.net/site/_config.yml
            Source: /home/therealone/git/therealone78.ddns.net/site
       Destination: /home/therealone/git/therealone78.ddns.net/site/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       Jekyll Feed: Generating feed for posts
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($spacing-unit, 2) or calc($spacing-unit / 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
40 │   margin-bottom: $spacing-unit / 2;
   │                  ^^^^^^^^^^^^^^^^^
   ╵
    ../../../../minima-2.5.1/_sass/minima/_base.scss 40:18                @import
    minima.scss 48:3                                                      @import
    /home/therealone/git/therealone78.ddns.net/site/assets/main.scss 1:9  root stylesheet
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($spacing-unit, 2) or calc($spacing-unit / 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
134 │   padding-left: $spacing-unit / 2;
    │                 ^^^^^^^^^^^^^^^^^
    ╵
    ../../../../minima-2.5.1/_sass/minima/_base.scss 134:17               @import
    minima.scss 48:3                                                      @import
    /home/therealone/git/therealone78.ddns.net/site/assets/main.scss 1:9  root stylesheet
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($spacing-unit, 2) or calc($spacing-unit / 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
189 │     padding-right: $spacing-unit / 2;
    │                    ^^^^^^^^^^^^^^^^^
    ╵
    ../../../../minima-2.5.1/_sass/minima/_base.scss 189:20               @content
    minima.scss 38:5                                                      media-query()
    ../../../../minima-2.5.1/_sass/minima/_base.scss 186:3                @import
    minima.scss 48:3                                                      @import
    /home/therealone/git/therealone78.ddns.net/site/assets/main.scss 1:9  root stylesheet
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($spacing-unit, 2) or calc($spacing-unit / 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
190 │     padding-left: $spacing-unit / 2;
    │                   ^^^^^^^^^^^^^^^^^
    ╵
    ../../../../minima-2.5.1/_sass/minima/_base.scss 190:19               @content
    minima.scss 38:5                                                      media-query()
    ../../../../minima-2.5.1/_sass/minima/_base.scss 186:3                @import
    minima.scss 48:3                                                      @import
    /home/therealone/git/therealone78.ddns.net/site/assets/main.scss 1:9  root stylesheet
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($spacing-unit, 3) or calc($spacing-unit / 3)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
244 │     padding: ($spacing-unit / 3) ($spacing-unit / 2);
    │               ^^^^^^^^^^^^^^^^^
    ╵
    ../../../../minima-2.5.1/_sass/minima/_base.scss 244:15               @import
    minima.scss 48:3                                                      @import
    /home/therealone/git/therealone78.ddns.net/site/assets/main.scss 1:9  root stylesheet
Warning: 6 repetitive deprecation warnings omitted.
Run in verbose mode to see all warnings.
                    done in 0.486 seconds.
 Auto-regeneration: enabled for '/home/therealone/git/therealone78.ddns.net/site'
    Server address: http://127.0.0.1:4000/
  Server running... press ctrl-c to stop.
```

**Comment by ashmaroli** [comment]
Hello @TheRealOne78,
The permission issue aside, deprecation warnings have been addressed via v2.5.2. Please run `bundle update minima` if you are using Minima via the gem package.
